src/main/java/seedu/internbuddy/model/company/NameContainsKeywordsPredicate.java

```diff
@@ -19,7 +19,11 @@ public NameContainsKeywordsPredicate(List<String> keywords) {
     @Override
     public boolean test(Company company) {
         return keywords.stream()
-                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(company.getName().fullName, keyword));
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(company.getName().fullName, keyword)
+                || StringUtil.containsWordIgnoreCase(company.getTagsString(), keyword)
+                || (company.getApplications() != null
+                        && (StringUtil.containsWordIgnoreCase(company.getAppNameString(), keyword)
+                        || StringUtil.containsWordIgnoreCase(company.getAppDescriptionString(), keyword))));
```

Comment:
I think this can be split into a few boolean expressions instead of one long expression (to be less confusing)